test/builtin/atomic/test_numbers.py

```diff
@@ -177,43 +178,87 @@ def test_n():
         check_evaluation(str_expr, str_expected)
 
 
-def test_accuracy():
-    for str_expr, str_expected in (
-        ("0`4", "0"),
-        ("Accuracy[0.0]", "15."),
-        ("Accuracy[0.000000000000000000000000000000000000]", "36."),
-        ("Accuracy[-0.0]", "15."),
-        # In WMA, this gives 36. Seems to be a rounding  issue
-        # ("Accuracy[-0.000000000000000000000000000000000000]", "36."),
-        ("1.0000000000000000 // Accuracy", "15."),
-        ("1.00000000000000000 // Accuracy", "17."),
+@pytest.mark.parametrize(
```

Comment:
I think Accuracy is based on computer hardware characteristics, right? If so we should add a `pytest.skipif`  when those hardware characteristics are not met.

Reply:
I used the native `sys.float_info`, so it would adjust to each architecture. If we discover one architecture where it fails, then we can handle it.

Reply:
Doesn't accuracy depend on MACHINE_PRECISION_VALUE and MIN_MACHINE_NUMBER? and those depend of mpmath values and you say that mpmath varies? 

I am okay with leaving out `pytest.mark.skipif`  _provided_ there is an assertion  on these underlying values. 

And then that way we will know that things have changed.

Recall we had a recent situation with reading bytes from a file that changed due to the endian-ness of the machine. 

I would feel differently if we were testing regularly using different CPU architectures and 32-bit machines versus 64-bit OS setups and so on. But we are not.

Reply:
One other thing here. There should be unit test precisely for the underlying assumptions that are in effect.

One purpose of _unit_ tests is exactly this. (There are other test like functional and integration tests). Our tests tend to be on the heavier side - integration and functional rather than on the smaller lighter, more precise unit tests.

Reply:
> Doesn't accuracy depend on MACHINE_PRECISION_VALUE and MIN_MACHINE_NUMBER? and those depend of mpmath values and you say that mpmath varies?

`MACHINE_PRECISION_VALUE` and `MIN_MACHINE_NUMBER` depends on numbers coming from the `sys` module, so it would work in any platform that runs Python. What should we catch?



> 
> I am okay with leaving out `pytest.mark.skipif` _provided_ there is an assertion on these underlying values.
> 
> And then that way we will know that things have changed.
> 
> Recall we had a recent situation with reading bytes from a file that changed due to the endian-ness of the machine.
> 
> I would feel differently if we were testing regularly using different CPU architectures and 32-bit machines versus 64-bit OS setups and so on. But we are not.

Reply:
I looked at the Python code, and most definitely these things  change per system - that is why it is in Python's `sys` module. In particular this seems to come from  the compiler's `<float.h>` values. 

I think we should have a function that is called by pytest.mark.skipif that returns False if any of MACHINE_EPSILON, MACHINE_PRECISION_VALUE, MIN_MACHINE_NUMBER, MAX_MACHINE_NUMBER are not the values that are assumed by the test.

Reply:
Ah! Now I understand. What I think would be better is to change the tests in a way that the connection with these numbers be reinforced.

Reply:
Yes, that is even better. And if you are up for it,  go for it! 

Maybe you can find a 32-bit OS around and see what values you get or try a Python made from a 32-bit C compiler. Or if there are 128-bit architectures or C compiler and see what those do. 

There might be github workflows CI images that have these properties of 32-bitness or 32-bitness compilers (or other than 64-bitness) and we could run a test using that image.

Reply:
And the same thing along the lines of different mpc versions.

Reply:
It occurs to me that is you feel confident in knowing how what MACHINE_EPSILON, etc change things, then in a test you could artificially set these to different values and make the Precision and Accuracy tests. 

It would however be nice to know what alternate values these possibly can take on.

Reply:
I think that with the new changes, the tests are robust across different platforms. The only thing that could go wrong is that certain platform does not have support for floating point numbers, but I guess that in such a case everything gets broken.